lib/enum.ml

```diff
@@ -57,6 +87,10 @@ module Str = struct
     in
     Ast_builder.Default.case ~lhs ~guard:None ~rhs
 
+  let from_string_constructor_cases ~loc ~raises constructors =
+    let constructor_details = List.map constructor_name_and_value constructors in
+    List.map (from_string_case_from_name_and_value ~loc ~raises) constructor_details
```

Comment:
I think you can make this easier to read with the following:

```ocaml
  constructors
  |> List.map constructor_name_and_value
  |> List.map (from_string_case_from_name_and_value ~loc ~raises)
```

Reply:
That is nicer :)